design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,242 @@
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the feature doc, before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+    - !variable
+        id: provider-uri # https://sts.windows.net/TENANT_ID
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+### Azure Application Identity
+
+To authenticate with Conjur, Azure-specific fields will need to be provided in the host annotations of the Conjur host identity for the Azure resource, 
+specifically subscription-id, resource-group, and user/system-assigned-identity. These fields will be used to confirm the identity of the Azure resource 
+sending the authentication request by comparing the Azure token information with the host annotations that are defined in the above policy.
+
+A Conjur host will be defined as follows with their annotations holding Azure-specific identification properties.
+
+```yaml
+- !policy 
+  id: azure-apps 
+  body: 
+  - !host 
+    id: test-app 
+    annotations: 
+    authn-azure/subscription-id: test-subscription 
+    authn-azure/resource-group: test-group 
+    authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+    authn-azure/system-assigned-identity: ie_qWCXhXxt1zIEsu4c7acQVGn4
+```
+ 
+#### Multiple-assigned identity for host - NOT FOR NOW
+
+We may allow hosts to be defined with both "authn-azure/user-assigned-identity" and "authn-azure/system-assigned-identity". This way such a host can 
+authenticate a token with a user-assigned id as well as a token with a system-assigned id. As a first step we will not develop - nor test - such a use-case.
+
+### Design
+
+#### Authentication Flow
+
+This flow assumes that the VM already exists in Azure, and that an authn-azure authenticator and a Conjur host were loaded.
+
+![Authn Azure Flow](authn-azure-flow.png)
+
+
+1. Azure VM sends the following request to Conjur:
+    ```yaml
+    POST https://<DAP-server-hostname>/authn-azure/<service-id>/<account>/<host-id>authenticate
+    Header 
+    
+    Content-Type: application/x-www-form-urlencoded
+    Body 
+    
+     token: "eyJhbGciOiJSUzI1NiIs......uTonCA"
```

Comment:
How is this token obtained?
From this endpoint `http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://management.azure.com/`

And if so are we parsing the `access_token` from this response and sending it to conjur?
If thats the case the access_token appears to be an azure specific JWT.
Since it is a JWT why not just call it `jwt`?

Reply:
Nevermind. I guess there is a difference between access_token and id_token mentioned here:
https://developer.okta.com/docs/guides/validate-access-tokens/overview/#access-tokens-vs-id-tokens